lib/active_record/connection_adapters/oracle_enhanced_adapter.rb

```diff
@@ -569,6 +569,10 @@ def materialized_views #:nodoc:
         select_values("SELECT LOWER(mview_name) FROM all_mviews WHERE owner = SYS_CONTEXT('userenv', 'current_schema')")
       end
 
+      def synonym_names #:nodoc:
+        select_values("SELECT LOWER(synonym_name) FROM all_synonyms where owner = SYS_CONTEXT('userenv', 'current_schema')")
```

Comment:
Current `OracleEnhancedAdapter` has already `synonyms` method which implemented in [`SchemeStatementsExt#synonyms`](https://github.com/rsim/oracle-enhanced/blob/abc596e7f8f78fa25cabc05b167cd8f0241e14d5/lib/active_record/connection_adapters/oracle_enhanced/schema_statements_ext.rb#L53-L58), so that `synonyms` method name conflicts with this, and this method returns `SynonymDefinition` objects. This is why I used `synonym_names` for this new method.

Instead of new method, how about this?
```ruby
def data_sources
  super | synonyms.map(&:name)
end
```